katello/katello-restore

```diff
@@ -31,39 +37,69 @@ end
 def reset_katello
   puts "Resetting Katello"
   `tar --selinux --overwrite --listed-incremental=/dev/null -xzf config_files.tar.gz -C /`
-  `foreman-installer --scenario katello --reset`
+  if @is_foremnan_proxy_content
+    `echo y | foreman-installer --scenario #{FOREMAN_PROXY_CONTENT} --reset --foreman-proxy-register-in-foreman false --disable-system-checks`
+  else
+    `foreman-installer --scenario katello --reset`
+  end
   puts "Done.\n"
 end
 
+def restore_psql_dumps
+  puts "Restoring postgresql dump files"
+  `service postgresql start`
+  `runuser - postgres -c "dropdb foreman"`
+  `runuser - postgres -c "dropdb candlepin"`
+  `runuser - postgres -c "psql -f #{DIR}/pg_globals.dump postgres 2>/dev/null"`
+  `runuser - postgres -c "pg_restore -C -d postgres #{DIR}/foreman.dump"`
+  `runuser - postgres -c "pg_restore -C -d postgres #{DIR}/candlepin.dump"`
+  `service postgresql stop`
+  puts "Done."
+end
+
+def migrate_pulp
+  puts "Migrating pulp databases"
+  `systemctl start mongod qpidd`
```

Comment:
Are we assured all services are stopped and then only these two are getting started? Newer Pulp requires that workers/celery beat and resource manager are running when migrating.